btree/btree.c

```diff
@@ -5529,8 +5448,11 @@ void m0_btree_destroy(struct m0_btree *arbor, struct m0_btree_op *bop,
 {
 	bop->bo_arbor = arbor;
 	bop->bo_tx    = tx;
-	bop->bo_seg   = arbor->t_desc->t_seg;
-
+	if (arbor->t_desc != NULL)
+		bop->bo_seg = arbor->t_desc->t_seg;
+	else
+		bop->bo_seg = NULL;
```

Comment:
Can you please explain the logic behind lines 5451-5454.

Reply:
I wanted to make the m0_btree_destroy code more robust so that it handles incorrect cases inside it. Thus, in the case of tree_close() -> tree_destroy(), it might be possible that t_desc has been freed earlier. Therefore, we might get a segmentation fault here. To avoid that and handle that case properly, I have added these lines.